### PR TITLE
feat(recovery): add doc on fault telorance mechanisms

### PR DESCRIPTION
**Pull request by baptistemesta**
This summarize how the platform handles incidents:

* Cluster for node failure
* Connectors replay for external service failure
* Retry + Recover to handle service outage like database

Relates to [BR-446](https://bonitasoft.atlassian.net/browse/BR-446)

### CODE REVIEW

File: md/fault-tolerance.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
+# Fault tolerance mechanisms
+
+Learn which mechanism allows the platform to be fault tolerant. 
+
+## Global fault tolerance strategy
+
+Multiple incidents can happen in a given system. We can have the database not responding anymore either because of a crash or 
+a network issue, we can have the JVM on which Bonita Platform run failing, or we can have external services not available. Each category of incident have a mechanism to recover from it.
```
Comment (danila-m) [inline md/fault-tolerance.md]:
```suggestion
a network issue, we can have the JVM on which Bonita Platform runs failing, or we can have external services not available. Each category of incident has a mechanism to recover from it.
```

File: md/fault-tolerance.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+* When dealing with machine wise failure like the JVM or physical host crashing, Bonita platform can be installed in a
```
Comment (danila-m) [inline md/fault-tolerance.md]:
```suggestion
* When dealing with a failure on the host machine, like the JVM or hardware crashing, Bonita platform can be installed in a
```

File: md/fault-tolerance.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+* When dealing with external services' outage called by [connectors](connectors-overview.md), they can be [replayed](tasks.md). 
+* When dealing with incident like unresponsive database, we have two mechanisms to handle these errors the Retry mechanism and 
+the Recovery mechanism 
+
+## Retry mechanism
+
+It is a reactive system, when an incident happens during the execution of a work, an exception is thrown. This mechanism
+analyses the exception and determine if the work should be retried automatically or not.
+
+When the exception is retryable, the platform re-queue the work with a delay. That can be done up to 10 times with a delay
```
Comment (danila-m) [inline md/fault-tolerance.md]:
```suggestion
When the exception is retryable, the platform queues the work again, with a delay. That can be done up to 10 times
```

File: md/fault-tolerance.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+increasing from 1 second to more than 1 hour by default.
```
Comment (danila-m) [inline md/fault-tolerance.md]:
```suggestion
with the delay increasing gradually from 1 second to more than 1 hour by default.
```

**Review by danila-m — APPROVED** [review]
Good job overall, remarks to improve grammar/make it more comprehensible.

File: md/fault-tolerance.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+# TYPE bonita_bpmengine_recovery_duration_seconds summary
+bonita_bpmengine_recovery_duration_seconds_active_count{tenant="1",} 0.0
+bonita_bpmengine_recovery_duration_seconds_duration_sum{tenant="1",} 0.0
+# HELP bonita_bpmengine_recovery_recovered_last_elements number of elements recovered
+# TYPE bonita_bpmengine_recovery_recovered_last_elements gauge
+bonita_bpmengine_recovery_recovered_last_elements{tenant="1",} 0.0
+# HELP bonita_bpmengine_recovery_recovered_total_elements_total Total number of elements recovered
+# TYPE bonita_bpmengine_recovery_recovered_total_elements_total counter
+bonita_bpmengine_recovery_recovered_total_elements_total{tenant="1",} 39768.0
+bonita_bpmengine_work_pending_works{tenant="1",} 0.0
```
Comment (passga) [inline md/fault-tolerance.md]: Delete line because these metrics does not related to the recovery mechanism.
```suggestion
```

**Review by passga — APPROVED** [review]
I have just one suggestion but the could be merged after.

**Pull request merged**